### DISCUSSION TITLE
Unable to project fields using batchGet

### DISCUSSION THREAD

**Discussion by kunalfleethawks** [sensedeep/dynamodb-onetable#212] [Q&A] [answered]
Hello Team,

We are performing following batchGet operation. For each batch query we are specifying projection fields but when batchGet executes it returns all fields specified in schema. Following is the sample code. 

As per [AWS Documentation ](https://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_BatchGetItem.html)we can specify projectionFields or fields (incase of this library) for each batch.

```
let batch = {}
await Account.get({id: accountId}, {batch, fields: 'name'})
await User.get({id: userId}, {batch,  fields: 'firstName' })
let results = await table.batchGet(batch)
```

Can you let us know how to achieve expected behaviour?

Thanks

**Comment by mobsense** [comment, marked as answer]
Fixed in 2.2.0.

See #213
